control-plane-operator/controllers/hostedcontrolplane/oauth/idp_convert.go

```diff
@@ -596,12 +682,16 @@ func isValidURL(rawurl string, optional bool) bool {
 	return u.Scheme == "https" && len(u.Host) > 0 && len(u.Fragment) == 0
 }
 
-func transportForCARef(ctx context.Context, kclient crclient.Client, namespace, name, key string) (http.RoundTripper, error) {
+func transportForCARef(ctx context.Context, kclient crclient.Client, namespace, name, key string, dialer konnectivityproxy.ProxyDialer) (http.RoundTripper, error) {
```

Comment:
When constructing this transport, if we are using the data plane dialer, we need to take into consideration the user's proxy configuration in hcp.configuration.proxy
to configure the proxy on the transport, pls see:
https://github.com/openshift/hypershift/blob/4a34cb844b3703811787afa7f0298cea38f52719/konnectivity-https-proxy/cmd.go#L73-L78
https://github.com/openshift/hypershift/blob/4a34cb844b3703811787afa7f0298cea38f52719/konnectivity-https-proxy/cmd.go#L113-L115

If a ca bundle is specified for the proxy, that CA bundle also needs to be added to the trusted roots of the transport.

Reply:
addressed!